bin/pycbc_live

```diff
@@ -872,8 +875,14 @@ with ctx:
                 bg_fn = os.path.join(args.output_background[1], bg_fn)
                 with h5py.File(bg_fn, 'w') as bgf:
                     for bg_ifos, bg_data, bg_time in bg_dists:
-                        ds = bgf.create_dataset(','.join(sorted(bg_ifos)),
-                                                data=bg_data, compression='gzip')
+                        if args.output_background_n_loudest:
+                            n_loudest = args.output_background_n_loudest
+                            assert (n_loudest > 0), "We can only store positive int loudest triggers."
+                            ds = bgf.create_dataset(','.join(sorted(bg_ifos)),
+                                                    data=-numpy.partition(-bg_data, n_loudest), compression='gzip')
```

Comment:
@bhooshan-gadre This doesn't look like it should work based on what the partition function does. I.E. it returns and array with the same length as its input. See https://numpy.org/doc/stable/reference/generated/numpy.partition.html

Reply:
Write. Modified accordingly. Forgot last part choosing n